src/components/views/dialogs/SlidingSyncOptionsDialog.tsx

```diff
@@ -27,89 +27,90 @@ import { useAsyncMemo } from "../../../hooks/useAsyncMemo";
         localTimeoutMs: 10 * 1000, // 10s
         prefix: "/_matrix/client/unstable/org.matrix.msc3575",
     });
-    logger.info("server natively support sliding sync OK");
+    logger.info("nativeSlidingSyncSupportCheck: server natively supports sliding sync");
 }
 
 /**
- * Check that the proxy url is in fact a sliding sync proxy endpoint and it is up.
+ * Check that the sliding sync endpoint is in fact a sliding sync endpoint and is up
  * @param endpoint The proxy endpoint url
- * @param hsUrl The homeserver url of the logged in user.
- * @throws if the proxy server is unreachable or not configured to the given homeserver
+ * @throws if the endpoint is unreachable
  */
-async function proxyHealthCheck(endpoint: string, hsUrl?: string): Promise<void> {
+async function slidingSyncHealthCheck(endpoint: string): Promise<void> {
     const controller = new AbortController();
     const id = window.setTimeout(() => controller.abort(), 10 * 1000); // 10s
     const res = await fetch(endpoint + "/client/server.json", {
         signal: controller.signal,
     });
     clearTimeout(id);
     if (res.status != 200) {
-        throw new Error(`proxyHealthCheck: proxy server returned HTTP ${res.status}`);
-    }
-    const body = await res.json();
-    if (body.server !== hsUrl) {
-        throw new Error(`proxyHealthCheck: client using ${hsUrl} but server is as ${body.server}`);
+        throw new Error(`slidingSyncHealthCheck: endpoint returned ${res.status}`);
     }
-    logger.info("sliding sync proxy is OK");
+    logger.info("slidingSyncHealthCheck: sliding sync endpoint exists");
 }
 
 export const SlidingSyncOptionsDialog: React.FC<{ onFinished(enabled: boolean): void }> = ({ onFinished }) => {
     const cli = MatrixClientPeg.safeGet();
-    const currentProxy = SettingsStore.getValue("feature_sliding_sync_proxy_url");
     const hasNativeSupport = useAsyncMemo(
         () =>
-            syncHealthCheck(cli).then(
+            nativeSlidingSyncSupportCheck(cli).then(
+                () => true,
+                () => false,
+            ),
+        [],
+        null,
+    );
+    const hasProxySupport = useAsyncMemo(
+        () =>
+            proxySlidingSyncSupportCheck(cli).then(
                 () => true,
                 () => false,
             ),
         [],
         null,
     );
 
-    let nativeSupport: string;
-    if (hasNativeSupport === null) {
-        nativeSupport = _t("labs|sliding_sync_checking");
+    let hasSupport: string;
+    if (hasNativeSupport === null && hasProxySupport === null) {
+        hasSupport = _t("labs|sliding_sync_checking");
     } else {
-        nativeSupport = hasNativeSupport
+        hasSupport = hasNativeSupport || hasProxySupport
             ? _t("labs|sliding_sync_server_support")
             : _t("labs|sliding_sync_server_no_support");
     }
 
-    const validProxy = withValidation<undefined, { error?: unknown }>({
```

Comment:
I'd like to reintroduce this check with `slidingSyncHealthCheck()`, but I'm not quite sure how to do it right now.

Reply:
Ditching the dialog out